sdk/log/batch.go

```diff
@@ -1,10 +1,13 @@
 // Copyright The OpenTelemetry Authors
 // SPDX-License-Identifier: Apache-2.0
 
+// Copyright 2009 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
```

Comment:
Can we move the (almost) copied code to `ring.go` and `ring_test.go`?
You could also add in comment(s) a hyperlink to the "original" source code.